### PR TITLE
Renamed vis structures

### PR DESCRIPTION
**Pull request by james-s-willis**

### CODE REVIEW

File: lib/core/bufferFactory.cpp [hunk 1]
```diff
@@ -1,10 +1,10 @@
 #include "bufferFactory.hpp"
 
 #include "Config.hpp"         // for Config
+#include "VisFrameView.hpp"   // for VisFrameView
```
Comment (andrerenard) [inline lib/core/bufferFactory.cpp:4]: Interesting that one of our auto formatters is case sensitive in it's sorting algorithm.  @nritsche any idea if that's easy to change?

File: lib/core/bufferFactory.cpp [hunk 2]
```diff
@@ -103,7 +103,7 @@ struct Buffer* bufferFactory::new_buffer(const string& type_name, const string&
         auto layout = VisFrameView::calculate_buffer_layout(num_elements, num_prod, num_ev);
         uint32_t frame_size = layout.first;
 
-        INFO_NON_OO("Creating visBuffer named {:s} with {:d} frames, frame size of {:d} and "
+        INFO_NON_OO("Creating VisFrameView named {:s} with {:d} frames, frame size of {:d} and "
```
Comment (andrerenard) [inline lib/core/bufferFactory.cpp:106]:
```suggestion
        INFO_NON_OO("Creating VisBuffer named {:s} with {:d} frames, frame size of {:d} and "
```
I think this should still be `VisBuffer`, the frame view is just a useful abstraction on the buffer.

File: lib/core/bufferFactory.cpp (hunk 1 above)
Comment (jrs65) [reply lib/core/bufferFactory.cpp:4]: > I'm not sure that it makes sense to rename `visBuffer` to `VisFrameView`. It seems to me in most places what we are really referring to is the `VisBuffer` not the frame view, the frame view is just constructed on top for convenience. @jrs65 any thoughts on this?

I don't mind either way. It's always been a bit odd to call it that, but there's no actual class called visBuffer.

However, for the moment I'd probably just leave it called visBuffer.*. It sounds like we're probably going to have a big reorg and rename at somepoint soon, so we might as well just do it in that.

**Comment by andrerenard** [comment]
@jrs65 Just to be clear here are you ok with the `visBuffer -> VisFrameView` changes as is, or did  you want @james-s-willis to switch to just `visBuffer -> VisBuffer` before merging?

**Comment by jrs65** [comment]
I think I would prefer to switch back to visBuffer.hpp etc, provided it's no more work than a `git mv` and a search and replace. If it's harder than that, don't bother.

File: lib/core/bufferFactory.cpp (hunk 1 above)
Comment (nritsche) [reply lib/core/bufferFactory.cpp:4]: > Interesting that one of our auto formatters is case sensitive in its sorting algorithm. @nritsche any idea if that's easy to change?

There's no option in clang-format to change that behavior.

**Comment by nritsche** [comment]
I'm a big fan of a header file containing one class and being named after it. In `visBuffer.hpp`, there's one class called `visFrameView`. Why call the file `VisBuffer.hpp`?

**Comment by jrs65** [comment]
> I'm a big fan of a header file containing one class and being named after it. In `visBuffer.hpp`, there's one class called `visFrameView`. Why call the file `VisBuffer.hpp`?

Originally it was because it contains two classes of fairly equal importance (`VisMetadata and VisFrameView`), which go together and make the VisBuffer interface.

I'm fine with the one class per file thing, but I think if we want to rename the file to `VisFrameView.[hc]pp` then we need to pull `VisMetadata` into it's own file too, otherwise when you look at the filename it kind of implies that you'll only find `VisFrameView` in it. At somepoint we should do both of these things, but I don't think it makes sense to do only one.

**Comment by james-s-willis** [comment]
I've renamed `VisFrameView.[hc]pp` back to `visBuffer.[hc]pp`.

File: lib/testing/FakeVis.hpp [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /*****************************************
```
Comment (jrs65) [inline lib/testing/FakeVis.hpp:1]: Pretty much all mentions in this file.

File: lib/utils/visBuffer.cpp [hunk 4]
```diff
@@ -119,7 +119,7 @@ std::string VisFrameView::summary() const {
     struct tm* tm = std::gmtime(&(std::get<1>(time).tv_sec));
 
     std::string s =
-        fmt::format("visBuffer[name={:s}]: freq={:d} dataset={} fpga_start={:d} time={:%F %T}",
+        fmt::format("VisFrameView[name={:s}]: freq={:d} dataset={} fpga_start={:d} time={:%F %T}",
```
Comment (jrs65) [inline lib/utils/visBuffer.cpp:122]: This one change.
```suggestion
        fmt::format("VisBuffer[name={:s}]: freq={:d} dataset={} fpga_start={:d} time={:%F %T}",
```

File: lib/testing/FakeVis.cpp [hunk 5]
```diff
@@ -141,7 +141,7 @@ void FakeVis::main_thread() {
 
         for (auto f : freq) {
 
-            DEBUG("Making fake visBuffer for freq={:d}, fpga_seq={:d}", f, fpga_seq);
+            DEBUG("Making fake VisFrameView for freq={:d}, fpga_seq={:d}", f, fpga_seq);
```
Comment (jrs65) [inline lib/testing/FakeVis.cpp:144]:
```suggestion
            DEBUG("Making fake VisBuffer for freq={:d}, fpga_seq={:d}", f, fpga_seq);
```

File: lib/stages/visWriter.hpp [hunk 6]
```diff
@@ -63,8 +63,8 @@
  *
```
Comment (jrs65) [inline lib/stages/visWriter.hpp:63]: This file.

File: lib/stages/visTruncate.hpp [hunk 7]
```diff
@@ -22,11 +22,11 @@
  *
```
Comment (jrs65) [inline lib/stages/visTruncate.hpp:22]: This file.

File: lib/stages/visTranspose.hpp [hunk 8]
```diff
@@ -39,8 +39,8 @@
  *
```
Comment (jrs65) [inline lib/stages/visTranspose.hpp:39]: This file.

File: lib/stages/visTransform.hpp [hunk 9]
```diff
@@ -1,6 +1,6 @@
 /*****************************************
```
Comment (jrs65) [inline lib/stages/visTransform.hpp:1]: This file.

File: lib/stages/visTransform.cpp [hunk 10]
```diff
@@ -115,7 +115,7 @@ void visTransform::main_thread() {
```
Comment (jrs65) [inline lib/stages/visTransform.cpp:115]: This file.

File: lib/stages/visTestPattern.hpp [hunk 11]
```diff
@@ -1,6 +1,6 @@
 /*****************************************
```
Comment (jrs65) [inline lib/stages/visTestPattern.hpp:1]: This file.

File: lib/stages/visRawReader.hpp [hunk 12]
```diff
@@ -34,8 +34,8 @@
  *
```
Comment (jrs65) [inline lib/stages/visRawReader.hpp:34]: This file.

File: lib/stages/visDebug.hpp [hunk 13]
```diff
@@ -22,8 +22,8 @@
  *
```
Comment (jrs65) [inline lib/stages/visDebug.hpp:22]: This file.

File: lib/stages/visCompression.hpp [hunk 14]
```diff
@@ -37,11 +37,11 @@
  *
```
Comment (jrs65) [inline lib/stages/visCompression.hpp:37]: This file.

File: lib/stages/visAccumulate.hpp [hunk 15]
```diff
@@ -31,7 +31,7 @@
```
Comment (jrs65) [inline lib/stages/visAccumulate.hpp:31]: This file.

File: lib/stages/timeDownsample.hpp [hunk 16]
```diff
@@ -28,11 +28,11 @@
  *
```
Comment (jrs65) [inline lib/stages/timeDownsample.hpp:28]: This file.

File: lib/stages/removeEv.hpp [hunk 17]
```diff
@@ -21,11 +21,11 @@
  *
```
Comment (jrs65) [inline lib/stages/removeEv.hpp:21]: This file.

File: lib/stages/prodSubset.hpp [hunk 18]
```diff
@@ -1,6 +1,6 @@
 /*****************************************
```
Comment (jrs65) [inline lib/stages/prodSubset.hpp:1]: This file.

File: lib/stages/freqSubset.hpp [hunk 19]
```diff
@@ -20,18 +20,18 @@
```
Comment (jrs65) [inline lib/stages/freqSubset.hpp:20]: This file.

File: lib/stages/freqSplit.hpp [hunk 20]
```diff
@@ -21,21 +21,21 @@
```
Comment (jrs65) [inline lib/stages/freqSplit.hpp:21]: This file.

File: lib/stages/eigenVis.hpp [hunk 21]
```diff
@@ -23,15 +23,15 @@
  *
```
Comment (jrs65) [inline lib/stages/eigenVis.hpp:23]: This file.

File: lib/stages/countCheck.hpp [hunk 22]
```diff
@@ -29,8 +29,8 @@
  *
```
Comment (jrs65) [inline lib/stages/countCheck.hpp:29]: This file.

File: lib/stages/applyGains.hpp [hunk 23]
```diff
@@ -45,11 +45,11 @@
  *
```
Comment (jrs65) [inline lib/stages/applyGains.hpp:45]: This file.

File: lib/stages/VisSharedMemWriter.hpp [hunk 24]
```diff
@@ -69,8 +69,8 @@
  *
```
Comment (jrs65) [inline lib/stages/VisSharedMemWriter.hpp:69]: This file.

File: lib/stages/RingMapMaker.hpp [hunk 25]
```diff
@@ -32,8 +32,8 @@
  *
```
Comment (jrs65) [inline lib/stages/RingMapMaker.hpp:32]: This file.

File: lib/stages/ReceiveFlags.hpp [hunk 26]
```diff
@@ -38,11 +38,11 @@
  *
```
Comment (jrs65) [inline lib/stages/ReceiveFlags.hpp:38]: This file.

File: lib/stages/InputSubset.hpp [hunk 27]
```diff
@@ -1,6 +1,6 @@
 /*****************************************
```
Comment (jrs65) [inline lib/stages/InputSubset.hpp:1]: This file.

File: lib/stages/EigenVisIter.hpp [hunk 28]
```diff
@@ -30,18 +30,18 @@
  *
```
Comment (jrs65) [inline lib/stages/EigenVisIter.hpp:30]: This file.

**Comment by james-s-willis** [comment]
@jrs65 I think this is ready to go now

**Pull request merged**